lib/package/s3/compileMethodsToS3.js

```diff
@@ -82,6 +82,25 @@ module.exports = {
       return `method.request.querystring.${http.key.queryStringParam}`
     }
 
+    if (http.key.timestamp) {
+      function zeroPad(n) {
+        return ('0' + n).slice(-2)
+      }
+      const now = new Date()
+      const timestamp = now
+        .toISOString()
+        .replace(/Z/, '')
+        .replace(/(T|:|-|\.)/g, '-')
+      const key = [
+        now.getFullYear(),
+        zeroPad(now.getMonth() + 1),
+        zeroPad(now.getDate()),
+        zeroPad(now.getHours()),
+        `${http.key.timestamp}-${timestamp}`
+      ].join('/')
+      return `'${key}'`
+    }
```

Comment:
Unfortunately, it's not gonna work the way you're imagining. This would create a static key that is equal to the timestamp of "when you ran `sls deploy`".

Reply:
Dang. Ok, so would the proposed change need to be in a different repo? Some of it anyway...

Reply:
We might be able to do this with custom VTL template. @erezrokah @horike37 do you know by any chance? If not, I can do some more digging.

Reply:
Umm 🤔 Not sure that. We may need to take a look more.

Reply:
https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html#util-template-reference

`$context.requestTime` or `$context.requestTimeEpoch` may make it possible.

Reply:
I'll look at it today and try to work out a working example

Reply:
It might be prudent to use a combination of current time & random value to cater for the scenario when the apigw is called more than once at the same moment.